aiida/storage/psql_dos/backend.py

```diff
@@ -170,18 +169,21 @@ def _clear(self) -> None:
 
         super()._clear()
 
-        session = self.get_session()
-
         with self.migrator_context(self._profile) as migrator:
 
-            with self.transaction():
-                for table_name in (
-                    'db_dbgroup_dbnodes', 'db_dbgroup', 'db_dblink', 'db_dbnode', 'db_dblog', 'db_dbauthinfo',
-                    'db_dbuser', 'db_dbcomputer'
-                ):
-                    session.execute(table(table_name).delete())
+            # First clear the contents of the database
+            with self.transaction() as session:
+
+                # Close the session otherwise the ``delete_tables`` call will hang as there will be an open connection
+                # to the PostgreSQL server and it will block the deletion and the command will hang.
```

Comment:
I understand this measure prevents connections opened by this particular process from interfering, but I guess there could be other connections open to the DB (daemon / REST API / verdi shell / ...)?

Is there a way we could check whether that is the case and exit with an error message instead of hanging?

Reply:
Not sure to be honest. Not sure if sqlalchemy provides an API to get all open connection to a database, even those that are managed by it. I doubt it. We would probably have to go straight to `psycopg` and directly execute a postgres command. Anyway, for now, this method is only being called during unit testing, so it is not that likely.